src/swap/action.rs

```diff
@@ -82,11 +110,13 @@ where
 {
 }
 
+/// Look for the event `E` in our "Memory".
```

Comment:
My mistake was to use a loaded term in programming like "memory". I was going for the concept of memory rather than a concrete thing. I will follow your advice.